lib/vdc/GridHelper.cpp

```diff
@@ -541,6 +549,117 @@ UnstructuredGridLayered *GridHelper::_make_grid_unstructured_layered(size_t ts,
+    for (int i = 0; i < bs1d.size(); i++) {
+        nblocks *= bmax[i] - bmin[i] + 1;
+        block_size *= bs1d[i];
+    }
+
+    vector<float *> xcblkptrs;
+    for (int i = 0; i < nblocks; i++) { xcblkptrs.push_back(blkvec[1] + i * block_size); }
+
+    // Block pointers for Y coordinates, which are always 1D
+    //
+    nblocks = 1;
+    block_size = 1;
+    for (int i = 0; i < bs1d.size(); i++) {
+        nblocks *= bmax[i] - bmin[i] + 1;
+        block_size *= bs1d[i];
+    }
+    vector<float *> ycblkptrs;
+    for (int i = 0; i < nblocks; i++) { ycblkptrs.push_back(blkvec[2] + i * block_size); }
+
+    // Block pointers for Z coordinates, which are always 2D
+    //
+    nblocks = 1;
+    block_size = 1;
+    for (int i = 0; i < bs.size(); i++) {
+        nblocks *= bmax[i] - bmin[i] + 1;
+        block_size *= bs[i];
+    }
+    vector<float *> zcblkptrs;
+    for (int i = 0; i < nblocks; i++) { zcblkptrs.push_back(blkvec[3] + i * block_size); }
+
+    // N.B. assumes blkvec contains contiguous blocks :-(
+    //
+    const int *vertexOnFace = conn_blkvec[0];
+    const int *faceOnVertex = conn_blkvec[1];
+    const int *faceOnFace = conn_blkvec.size() == 3 ? conn_blkvec[2] : NULL;
+
+    vector<size_t> vertexDims1D = {vertexDims[0]};
+    vector<size_t> faceDims1D = {faceDims[0]};
+    vector<size_t> edgeDims1D;
+    if (edgeDims.size()) { edgeDims1D.push_back(edgeDims[0]); }
+
+    UnstructuredGridCoordless xug(vertexDims1D, faceDims1D, edgeDims1D, bs1d, xcblkptrs, 2, vertexOnFace, faceOnVertex, faceOnFace, location, maxVertexPerFace, maxFacePerVertex, vertexOffset,
+                                  faceOffset);
+
+    UnstructuredGridCoordless yug(vertexDims1D, faceDims1D, edgeDims1D, bs1d, ycblkptrs, 2, vertexOnFace, faceOnVertex, faceOnFace, location, maxVertexPerFace, maxFacePerVertex, vertexOffset,
+                                  faceOffset);
+
+    UnstructuredGridCoordless zug(vertexDims, faceDims, edgeDims, bs, zcblkptrs, 3, vertexOnFace, faceOnVertex, faceOnFace, location, maxVertexPerFace, maxFacePerVertex, vertexOffset, faceOffset);
+
+    string qtr_key = _getQuadTreeRectangleKey(ts, level, lod, cvarsinfo, bmin, bmax);
+
+    // Try to get a shared pointer to the QuadTreeRectangle from the
+    // cache. If one does not exist the Grid class will make one. We use
+    // a shared pointer so that we can cache it for use by other Grid
+    // classes. This a peformance optimization, necessary be creating
+    // a QuadTreeRectangle is expensive.
+    //
+    std::shared_ptr<const QuadTreeRectangle<float, size_t>> qtr = _qtrCache.get(qtr_key);
+
+    UnstructuredGrid3D *g = new UnstructuredGrid3D(vertexDims, faceDims, edgeDims, bs, blkptrs, vertexOnFace, faceOnVertex, faceOnFace, location, maxVertexPerFace, maxFacePerVertex, vertexOffset,
+                                                   faceOffset, xug, yug, zug, qtr);
+
+    // No QuadTreeRectangle in cache. So get shared pointer for one created
```

Comment:
Remove commented out code. A quadtree wouldn't work for 3D data anyway.

Reply:
Done.